cmd/pi-strings-go/strings.go

```diff
@@ -0,0 +1,191 @@
+package main
+
+import (
+	"sort"
+	"strings"
+	"unicode"
+	"unicode/utf8"
+
+	stringspb "github.com/lapsang-boys/pippi/proto/strings"
+	"golang.org/x/text/encoding"
+	"golang.org/x/text/encoding/traditionalchinese"
+	textunicode "golang.org/x/text/encoding/unicode"
+	"golang.org/x/text/encoding/unicode/utf32"
+	"golang.org/x/text/transform"
+)
+
+// extractStrings extracts printable strings with a minimum number of characters
+// from the given binary.
+func extractStrings(buf []byte, minLength int) []*stringspb.StringInfo {
+	c := make(chan []*stringspb.StringInfo)
+	encs := []struct {
+		enc stringspb.Encoding
+		dec *encoding.Decoder
+	}{
+		// * UTF-8
+		{
+			enc: stringspb.Encoding_UTF8,
+			dec: textunicode.UTF8.NewDecoder(),
+		},
+		// * UTF-16
+		//    - big endian
+		{
+			enc: stringspb.Encoding_UTF16BigEndian,
+			dec: textunicode.UTF16(textunicode.BigEndian, textunicode.IgnoreBOM).NewDecoder(),
+		},
+		//    - big endian, with BOM
```

Comment:
Are we really expecting BOM prefixed strings inside a binary?

Reply:
What it comes to BOMs in UTF:

![foo2](https://user-images.githubusercontent.com/1414531/66701926-0e786700-ed02-11e9-928b-7c6110645939.gif)

Reply:
:heart_eyes: 

Hahha, well it doesn't hurt, but I'd be surprised if they find anything. We should probably add a metrics service for everything we do (opt-in), so we can evaluate all the features we add